learn/best_practices/type_casting.md

```diff
@@ -0,0 +1,494 @@
+---
+layout: book
+title: Type Casting in Callbacks
+permalink: /learn/best_practices/type_casting
+---
+
+There are essentially five different ways to do that, each with its own
+advantages and disadvantages.
+
+The methods I, II and V can be used both in C and Fortran. The methods
+III and IV are only available in Fortran. The method VI is obsolete and
+should not be used.
+
+Work Arrays
```

Comment:
```suggestion
I) Work Arrays
```

Reply:
I did remove the explicit numbering, because it is kind of cumbersome to maintain. We should rather create references inside the chapter and link if this is possible with our current static side generator.